### PR TITLE
Fix inconsistencies, enforce naming conventions

### PR DESCRIPTION
**Pull request by cristian-recoseanu**
Fixed some inconsistencies, formatting and tackled points:
7 - Cleanup all ncMethodResult variants
9 - Enum naming convention
10 - Class, Methods, Events naming convention

Also identified the following:
* Missing NcTimePtp
* Missing NcBitset

### CODE REVIEW

File: docs/idl/NC-Framework.webidl [hunk 1]
```diff
@@ -450,212 +432,218 @@ $macro(BlockDatatypes)
...
+		"Disconnected",		// 2 Not connected to a stream
+		"ConnectionError"	// 3 Connected but broken
 	};
 	
-	interface ncPayloadStatus {				// Received payload status
-		"Undefined",								// 0	This is the value when there's no connection.
-		"PayloadOK",								// 1	Payload type is one we know about and the PDU is well-formed
-		"PayloadFormatUnsupported",					// 2	Payload is not one we know about
-		"PayloadError",								// 3	Some kind of error has occurred
+	interface NcPayloadStatus {
```
Comment (JeffBerryman) [inline docs/idl/NC-Framework.webidl]: I think this should be an enum, not an interface.  It appears to be an error in the original file.

File: docs/idl/NC-Framework.webidl [hunk 2]
```diff
@@ -5,11 +5,11 @@ $macro(HeaderComments)
 	//	NC-Framework classes and datatypes
```
Comment (JeffBerryman) [inline docs/idl/NC-Framework.webidl:5]: Maybe somewhere in these opening comments we should document the naming conventions.  Just an idea.

**Review by JeffBerryman — CHANGES_REQUESTED** [review]

**Comment by JeffBerryman** [comment]
I found one tiny mistake (see my first comment) that was inherited from the original.  It might be good to fix it now.

File: docs/idl/NC-Framework.webidl (hunk 2 above)
Comment (cristian-recoseanu) [reply docs/idl/NC-Framework.webidl:5]: We will tackle as a separate point by adding wording in the Markdown.

**Comment by cristian-recoseanu** [comment]
Reviewed on the call on the 19th of April 2022.
Missing NcTimePtp and missing NcBitset will be dealt with separately.

**Pull request merged**